pkg/controller/keycloakrealm/keycloakrealm_reconciler_test.go

```diff
@@ -0,0 +1,91 @@
+package keycloakrealm
+
+import (
+	"github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	"github.com/keycloak/keycloak-operator/pkg/common"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func getDummyRealm() *v1alpha1.KeycloakRealm {
+	return &v1alpha1.KeycloakRealm{
+		Spec: v1alpha1.KeycloakRealmSpec{
+			BrowserRedirectorIdentityProvider: "",
+			InstanceSelector: &v1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "keycloak",
+				},
+			},
+			KeycloakApiRealm: &v1alpha1.KeycloakApiRealm{
+				ID:          "dummy",
+				Realm:       "dummy",
+				Enabled:     true,
+				DisplayName: "dummy",
+				Users: []*v1alpha1.KeycloakApiUser{
+					{
+						ID:        "dummy",
+						UserName:  "dummy",
+						FirstName: "dummy",
+						LastName:  "dummy",
+						Enabled:   true,
+						Credentials: []v1alpha1.KeycloakCredential{
+							{
+								Type:      "password",
+								Value:     "password",
+								Temporary: false,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getDummyState() *common.RealmState {
+	return &common.RealmState{
+		Realm:            nil,
+		RealmUserSecrets: nil,
+		Context:          nil,
+		Keycloak:         nil,
+	}
+}
+
+func TestKeycloakRealmReconciler_Reconcile(t *testing.T) {
```

Comment:
Could you please structure tests using "given/when/then" pattern. It makes them much easier to read. You may find more here http://www.blog.j-labs.pl/2017/02/Given-When-Then-pattern-in-unit-tests